roles/satellite/tasks/internals/activationkey-hostgroups-deprovision.yml

```diff
@@ -4,25 +4,24 @@
     password: "{{ satellite_pwd }}"
     server_url: "{{ satellite_url }}"
     validate_certs: "{{ satellite_verify_ssl }}"
-
-    name: "{{ g_prod_hostgroup }}"
+    name: "{{ prod_hostgroup }}"
     organization: "{{ satellite_organization }}"
     organizations:
     - "{{ satellite_organization }}"
-    domain: "{{ hostvars[groups['prod'][0]]['domain'] if g_prod_domain_is_groupvar else prod_domain | default(omit) }}"
-    architecture: "{{ hostvars[groups['prod'][0]]['architecture'] if g_prod_architecture_is_groupvar else prod_architecture | default(omit) }}"
-    ptable: "{{ hostvars[groups['prod'][0]]['partition_table'] if g_prod_partition_table_is_groupvar else prod_partition_table | default(omit) }}"
-    operatingsystem: "{{ hostvars[groups['prod'][0]]['operatingsystem'] if g_prod_operatingsystem_is_groupvar else prod_operatingsystem | default(omit) }}"
-    medium: "{{ hostvars[groups['prod'][0]]['provision_installation_media'] if g_prod_medium_is_groupvar else prod_provision_installation_media | default(omit) }}"
-    content_source: "{{ hostvars[groups['prod'][0]]['foreman_capsule'] if g_prod_foreman_capsule_is_groupvar else prod_foreman_capsule | default(omit) }}"
-    lifecycle_environment: "{{ g_prod_lifecycleenvironment }}"
-    parent: "{{ hostvars[groups['prod'][0]]['hostgroup_parent'] if g_prod_hostgroup_parent_is_groupvar else prod_hostgroup_parent | default(omit) }}"
-    pxe_loader: "{{ hostvars[groups['prod'][0]]['pxe_loader'] if g_prod_pxe_loader_is_groupvar else prod_pxe_loader | default(omit) }}"
+    domain: "{{ prod_domain | default(omit) }}"
+    architecture: "{{ prod_architecture | default(omit) }}"
+    ptable: "{{ prod_partition_table | default(omit) }}"
+    operatingsystem: "{{ prod_operatingsystem | default(omit) }}"
+    medium: "{{ prod_provision_installation_media | default(omit) }}"
+    content_source: "{{ prod_foreman_capsule | default(omit) }}"
+    lifecycle_environment: "{{ prod_lifecycleenvironment }}"
+    parent: "{{ prod_hostgroup_parent | default(omit) }}"
+    pxe_loader: "{{ prod_pxe_loader | default(omit) }}"
     locations: "{{ locations.keys() | list }}"
     parameters: "{{ [satellite_hostgroup_parameters_defaults, hostgroup_parameters | default([])] | community.general.lists_mergeby('name') }}"
     content_view: "{{ deploymentcontentview }}"
     state: absent
-  when: g_prod_lifecycleenvironment_defined and g_prod_activationkey_defined and g_prod_activationkey_defined
+  when: prod_hostgroup is defined
```

Comment:
You missed the when `prod_lifecycleenvironment is defined` and `prod_activationkey is defined`